src/contracts/identity.ts

```diff
@@ -67,6 +72,25 @@ export const createBeaconProxyWithOwner = async (
   return contract.createBeaconProxyWithOwner(beacon, owner);
 };
 
+/**
+ * Create a new identity and register it to a handle to get a new DSNP Id.
+ * This will create and register a new beacon proxy identity.
+ *
+ * @param userAddress User's public key address
+ * @param handle The string handle to register
+ * @returns The contract Transaction
+ */
+export const createAndRegisterBeaconProxy = async (
+  userAddress: EthereumAddress,
+  handle: string
+): Promise<ContractTransaction> => {
+  const beaconFactory = await getBeaconFactoryContract();
+  const beaconAddr = await getBeaconAddress();
+  const { signer } = getConfig();
```

Comment:
do we want to pass config options?

Reply:
This line is no longer needed. Configuration options are used by some of the methods lower down. We're currently not passing in config for other methods in identity, so I'm going to stick to the pattern. We may decide to change all of them later.